### DISCUSSION TITLE
Any suggestions to handle different variations?

### DISCUSSION THREAD

**Discussion by lw78943** [iqtree/iqtree2#220] [Q&A] [answered]
Dear IQTREE team,

Thank you for sharing this great tool! It's very powerful and useful. I have constructed a phylogenic tree using SNPs. I want to try to construct a phylogenic tree using SNPs, Indels and structure variations together. I have VCF file that contains all of the Indels and structure variations. What do you recommend me to do with these raw variations before running iqtree? For SNPs, I have filtered missing data, singleton and did LD pruning, should I do similar filter process to these variations? For running iqtree, I have read the manual, but still confused how to run it with these different variations. Any suggestions are appreciated. Thank you so much for your generous support and help!

Bests,

Li

**Comment by roblanf** [comment, marked as answer]
Hi @lw78943,

For indels and structural variants, you'd need to code these as morphological characters, and then decide which model you want to use. For example, you might code a single indel as a binary character (present/absent in each sequence) then use a binary model. The available models in IQ-TREE for morphological data are listed under "Binary and morphological models" here:

http://www.iqtree.org/doc/Substitution-Models

I'm not all that familiar with best practice here, so it would be worth reading recent papers to make sure that you are fitting sensible models for your data type. 

And yes - sensible filtering should still be useful for indels and structural variants. Once you code each as a single morphological character, the same I.I.D. assumptions that apply sequence data also apply to morphological data.

Hope some of that helps,

Rob
